src/answers/services/answers.service.ts

```diff
@@ -228,4 +231,60 @@ export class AnswersService extends BaseService<
       }
     });
   }
+
+  async getUrls(answer: AnswerDocument): Promise<AnswerDocument> {
+    const responses = answer.responses;
+    for await (const response of responses) {
+      if (response.value && this.isUrlArrayType(response.value)) {
+        const newArray: IAnswerFile[] = [];
+        for await (const file of response.value) {
+          newArray.push({
+            url: await this.s3Service.generatePresignedUrl({
+              key: file.url,
+            }),
+            isPublic: file.isPublic,
+          });
+        }
+        response.value = newArray;
```

Comment:
Can we clean this into a map
```suggestion
        const presignedUrlPromises = response.value.map(async (file: IAnswerFile) => ({
          url: await this.s3Service.generatePresignedUrl({
            key: file.url,
          }),
          isPublic: file.isPublic,
        });
        response.value = await Promise.all(presignedUrlPromises);
```